code/modules/flufftext/Hallucination.dm

```diff
@@ -346,82 +397,122 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 	QDEL_NULL(fakebroken)
 	QDEL_NULL(fakerune)
 	QDEL_NULL(bubblegum)
+	STOP_PROCESSING(SSfastprocess, src)
 	return ..()
 
 /datum/hallucination/battle
+	var/battle_type
+	var/iterations_left
+	var/hits = 0
+	var/next_action = 0
+	var/turf/source
 
-/datum/hallucination/battle/New(mob/living/carbon/C, forced = TRUE, battle_type)
-	set waitfor = FALSE
+/datum/hallucination/battle/New(mob/living/carbon/C, forced = TRUE, new_battle_type)
 	..()
-	var/turf/source = random_far_turf()
-	if(!battle_type)
-		battle_type = pick("laser","disabler","esword","gun","stunprod","harmbaton","bomb")
+
+	source = random_far_turf()
+
+	battle_type = battle_type || pick("laser", "disabler", "esword", "gun", "stunprod", "harmbaton", "bomb")
```

Comment:
While this is a line shorter, || is a rather out of the way operator, and honestly I'd prefer if we didn't use it too much

Reply:
mso gave you the triple emote of shame...........

Reply:
will do it anyway if you want though

Reply:
If a reasonably experienced coder needs to ask what something does and all it's doing is saving us a line in relatively simple code, we shouldn't use it. I don't hate the operator universally, but I don't think this is a good place for it